src/main/kotlin/uk/gov/dluhc/printapi/service/temporarycertificate/ExplainerPdfTemplateDetailsFactory.kt

```diff
@@ -1,65 +1,58 @@
-    @Value("\${temporary-certificate.explainer-pdf.welsh.placeholder.ero-address-line3}") private val eroAddressLine3CyPlaceholder: String,
-    @Value("\${temporary-certificate.explainer-pdf.welsh.placeholder.ero-address-line4}") private val eroAddressLine4CyPlaceholder: String,
-    @Value("\${temporary-certificate.explainer-pdf.welsh.placeholder.ero-address-postcode}") private val eroAddressPostcodeCyPlaceholder: String,
-    @Value("\${temporary-certificate.explainer-pdf.welsh.placeholder.ero-email}") private val eroEmailAddressCyPlaceholder: String,
-    @Value("\${temporary-certificate.explainer-pdf.welsh.placeholder.ero-phone}") private val eroPhoneNumberCyPlaceholder: String,
+    private val pdfTemplateProperties: TemporaryCertificateExplainerPdfTemplateProperties
 ) {
 
     fun getTemplateDetails(gssCode: String, eroDto: EroDto): TemplateDetails {
         return if (isWalesCode(gssCode)) {
-            TemplateDetails(pdfTemplateWelsh, getWelshTemplatePlaceholders(eroDto))
+            TemplateDetails(
+                pdfTemplateProperties.welsh.path,
+                getTemplatePlaceholders(eroDto.welshContactDetails!!, pdfTemplateProperties.welsh.placeholder)
+            )
         } else {
-            TemplateDetails(pdfTemplateEnglish, getEnglishTemplatePlaceholders(eroDto))
+            TemplateDetails(
+                pdfTemplateProperties.english.path,
+                getTemplatePlaceholders(eroDto.englishContactDetails, pdfTemplateProperties.english.placeholder)
+            )
         }
     }
 
-    private fun getWelshTemplatePlaceholders(eroDto: EroDto): Map<String, String> {
-        val eroContactDetails = eroDto.welshContactDetails!!
-        return mapOf(
-            eroNameCyPlaceholder to eroContactDetails.name,
-            eroAddressLine1CyPlaceholder to eroContactDetails.address.property.orEmpty(),
-            eroAddressLine2CyPlaceholder to eroContactDetails.address.street,
-            eroAddressLine3CyPlaceholder to eroContactDetails.address.town.orEmpty(),
-            eroAddressLine4CyPlaceholder to eroContactDetails.address.area.orEmpty(),
-            eroAddressPostcodeCyPlaceholder to eroContactDetails.address.postcode,
-            eroEmailAddressCyPlaceholder to eroContactDetails.emailAddress,
-            eroPhoneNumberCyPlaceholder to eroContactDetails.phoneNumber,
-        )
-    }
+    private fun getTemplatePlaceholders(
+        eroContactDetails: EroContactDetailsDto,
+        pdfPlaceholders: Placeholder
+    ): Map<String, String> {
+        val orderedNonBlankValues = listOfNotNull(
+            eroContactDetails.name,
+            eroContactDetails.address.property,
+            eroContactDetails.address.street,
+            eroContactDetails.address.town,
+            eroContactDetails.address.area,
+            eroContactDetails.address.postcode,
+            eroContactDetails.emailAddress,
+            eroContactDetails.phoneNumber
+        ).filter { it.isNotBlank() }
```

Comment:
That's the intention.
Previously there was a static relationship between a piece of data and the form field that it should populate.  However, now we have 8 numbered form fields and the data is assigned dynamically so that form fields are populated top down without any left blank.  So if we have 8 pieces of data all 8 fields are populated, if there is only 5 pieces of data, this is used to populate the first 5 form fields with the 3 at the bottom left blank.  Basically so don't have blank lines in the generated document.